Framework/Algorithms/src/CalcCountRate.cpp

```diff
@@ -0,0 +1,653 @@
+#include "MantidAlgorithms/CalcCountRate.h"
+
+#include "MantidKernel/PropertyWithValue.h"
+#include "MantidKernel/BoundedValidator.h"
+#include "MantidKernel/TimeSeriesProperty.h"
+#include "MantidKernel/ListValidator.h"
+#include "MantidKernel/UnitFactory.h"
+#include "MantidKernel/MandatoryValidator.h"
+#include "MantidKernel/make_unique.h"
+
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/Axis.h"
+#include "MantidAPI/NumericAxis.h"
+#include "MantidAPI/WorkspaceFactory.h"
+
+#include "MantidDataObjects/Workspace2D.h"
+#include <numeric>
+
+namespace Mantid {
+namespace Algorithms {
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(CalcCountRate)
+
+//----------------------------------------------------------------------------------------------
+
+/// Algorithms name for identification. @see Algorithm::name
+const std::string CalcCountRate::name() const { return "CalcCountRate"; }
```

Comment:
That's the tricky one. As you may notice, initial name was Calculate but changed to Calc as second thought because its indeed too many Calculate algorithms. 
And according to Mantid conventions http://www.mantidproject.org/Mantid_Standards#Algorithms, the algorithms which do similar things should be named similarly, so the second suggestion is not comply with standards. 

I have no tough opinion on that, but the change to Calc was the second thought. I am glad to accept this change if you indeed think this is incorrect after the second thought.